org.adempiere.base/src/org/adempiere/model/GenericPO.java

```diff
@@ -132,6 +146,604 @@ protected int get_AccessLevel() {
+	/** In Progress = IP */
+	public static final String DOCSTATUS_InProgress = "IP";
+	/** Waiting Payment = WP */
+	public static final String DOCSTATUS_WaitingPayment = "WP";
+	/** Waiting Confirmation = WC */
+	public static final String DOCSTATUS_WaitingConfirmation = "WC";
+	/**
+	 * Set Document Status.
+	 *	@param DocStatus The current status of the document
+	 */
+	public void setDocStatus(String DocStatus) {
+		columnExists(COLUMNNAME_DocStatus, true);
+		set_Value (COLUMNNAME_DocStatus, DocStatus);
+	}
+
+	/**
+	 * Get Document Status.
+	 * @return The current status of the document
+	 */
+	public String getDocStatus() {
+		columnExists(COLUMNNAME_DocStatus, true);
+		return (String)get_Value(COLUMNNAME_DocStatus);
+	}
+
+	/**************************************************************************
+	 * 	Process document
+	 *	@param processAction document action
+	 *	@return true if performed
+	 */
+	public boolean processIt (String processAction) {
+		m_processMsg = null;
+		DocumentEngine engine = new DocumentEngine (this, getDocStatus());
+		return engine.processIt (processAction, getDocAction());
+	}	//	process
+
+	/**	Process Message 			*/
+	private String		m_processMsg = null;
+	/**	Just Prepared Flag			*/
+	private boolean		m_justPrepared = false;
+
+	/**
+	 * 	Unlock Document.
+	 * 	@return true if success
+	 */
+	public boolean unlockIt()
+	{
+		if (log.isLoggable(Level.INFO)) log.info("unlockIt - " + toString());
+		setProcessing(false);
+		return true;
+	}	//	unlockIt
+
+	/** Set Process Now.
+		@param Processing Process Now
+	 */
+	public void setProcessing (boolean Processing)
+	{
+		columnExists(COLUMNNAME_Processing, true);
+		set_Value (COLUMNNAME_Processing, Boolean.valueOf(Processing));
+	}
+
+	/** DocAction AD_Reference_ID=135 */
+	public static final int DOCACTION_AD_Reference_ID=135;
+	/** Complete = CO */
+	public static final String DOCACTION_Complete = "CO";
+	/** Approve = AP */
+	public static final String DOCACTION_Approve = "AP";
+	/** Reject = RJ */
+	public static final String DOCACTION_Reject = "RJ";
+	/** Post = PO */
+	public static final String DOCACTION_Post = "PO";
+	/** Void = VO */
+	public static final String DOCACTION_Void = "VO";
+	/** Close = CL */
+	public static final String DOCACTION_Close = "CL";
+	/** Reverse - Correct = RC */
+	public static final String DOCACTION_Reverse_Correct = "RC";
+	/** Reverse - Accrual = RA */
+	public static final String DOCACTION_Reverse_Accrual = "RA";
+	/** Invalidate = IN */
+	public static final String DOCACTION_Invalidate = "IN";
+	/** Re-activate = RE */
+	public static final String DOCACTION_Re_Activate = "RE";
+	/** <None> = -- */
+	public static final String DOCACTION_None = "--";
+	/** Prepare = PR */
+	public static final String DOCACTION_Prepare = "PR";
+	/** Unlock = XL */
+	public static final String DOCACTION_Unlock = "XL";
+	/** Wait Complete = WC */
+	public static final String DOCACTION_WaitComplete = "WC";
+	/**

```

Comment:
same as above, should use constants from DocAction interface instead.

Reply:
done